### PR TITLE
feat: resource definition criteria resource

### PR DESCRIPTION
**Pull request by johanneswuerbach**
Add a resource definition criteria resource that allows managing resource definition criteria separately.

While this is being tested you need to explicitly ignore any criteria changes in the resource definition as documented in the example.

### CODE REVIEW

File: internal/provider/resource_definition_criteria_resource.go [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+	var data *ResourceDefinitionCriteriaResourceModel
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	force := false
```
Comment (nickhumanitec) [inline internal/provider/resource_definition_criteria_resource.go:245]: :eyes: I would probably add this as input, and by default false?

File: internal/provider/resource_definition_criteria_resource.go [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+
+	if res.Criteria == nil {
+		resp.Diagnostics.AddWarning("Resource definition criteria not found", fmt.Sprintf("The resource definition criteria (%s) was deleted outside Terraform", data.ID.ValueString()))
+		resp.State.RemoveResource(ctx)
+		return
+	}
+
+	criteria := *res.Criteria
+
+	found := false
```
Comment (nickhumanitec) [inline internal/provider/resource_definition_criteria_resource.go:209]: this is just an observation. Looking at the API, you look into the criteria{} to identify if this is there or not.
 I guess we need an endpoint later on :eyes: all good

File: internal/provider/resource_definition_criteria_resource.go [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+	force := false
+	httpResp, err := r.client().DeleteOrgsOrgIdResourcesDefsDefIdCriteriaCriteriaIdWithResponse(ctx, r.orgId(), data.ResourceDefinitionID.ValueString(), data.ID.ValueString(), &client.DeleteOrgsOrgIdResourcesDefsDefIdCriteriaCriteriaIdParams{
+		Force: &force,
+	})
+	if err != nil {
+		resp.Diagnostics.AddError(HUM_CLIENT_ERR, fmt.Sprintf("Unable to delete definition criteria, got error: %s", err))
+		return
+	}
+
+	if httpResp.StatusCode() != 204 {
```
Comment (nickhumanitec) [inline internal/provider/resource_definition_criteria_resource.go:254]: This one, if possible, for 409 errors in general, add a timeout.

it is possible resources are being de provisioned while trying deleting this, and will work in a few min, so a retry is ok. Say default timeout is 3 min.

 If this add a lot of complexity now, could be done later.

**Review by nickhumanitec — CHANGES_REQUESTED** [review]
wonderful, looks good overall, a couple of comments but nothing major

File: internal/provider/resource_definition_criteria_resource.go (hunk 2 above)
Comment (johanneswuerbach) [reply internal/provider/resource_definition_criteria_resource.go:227]: Yes 👍, but that will take a bit more time.

**Pull request merged**